### PR TITLE
Added logging support for object removal

### PR DESCRIPTION
**Pull request by kirkeaton**

### CODE REVIEW

File: java_src/foam/nanos/audit/AuditDAO.java [hunk 1]
```diff
@@ -38,10 +38,20 @@ private String formatMessage(FObject currentValue, FObject newValue) {
...
     logger.info("CHANGE", obj.getClassInfo().getId(), user.getId(), formatMessage(current, obj));
     return super.put(obj);
   }
+
+  @Override
+  public FObject remove(FObject obj) {
+    // TODO: use context-oriented context when available.
+    User user = (User) getX().get("user");
+    NanoLogger logger = (NanoLogger) getX().get("logger");
+    logger.info("REMOVE", obj.getClassInfo().getId(), user.getId(), obj);
```
Comment (kgrgreer) [inline java_src/foam/nanos/audit/AuditDAO.java]: What does the toString() of obj return? Better to use JSON support to format obj?

Comment (kirkeaton) [reply java_src/foam/nanos/audit/AuditDAO.java]: Made the changes.

File: java_src/foam/nanos/audit/AuditDAO.java [hunk 2]
```diff
@@ -38,10 +40,22 @@ private String formatMessage(FObject currentValue, FObject newValue) {
...
     return super.put(obj);
   }
+
+  @Override
+  public FObject remove(FObject obj) {
+    // TODO: use context-oriented context when available.
+    User user = (User) getX().get("user");
+    NanoLogger logger = (NanoLogger) getX().get("logger");
+    StringBuilder sb = new StringBuilder();
+    new Outputter().output(sb, obj);
```
Comment (kgrgreer) [inline java_src/foam/nanos/audit/AuditDAO.java]: Can you reuse the Outputter to avoiding creating a new one for each remove()?

File: java_src/foam/nanos/audit/AuditDAO.java [hunk 3]
```diff
@@ -38,10 +40,22 @@ private String formatMessage(FObject currentValue, FObject newValue) {
...
+    logger.info("REMOVE", obj.getClassInfo().getId(), user.getId(), sb.toString());
```
Comment (kgrgreer) [inline java_src/foam/nanos/audit/AuditDAO.java]: Just pass 'sb', not 'sb.toString()'. The logger will call toString() on your arguments anyway, and this way, if this log message is filtered out, then you saved the effort.

File: java_src/foam/nanos/audit/AuditDAO.java (hunk 2 above)
Comment (kirkeaton) [reply java_src/foam/nanos/audit/AuditDAO.java]: Sure thing

**Pull request merged**